src/lib/containers/widgets/query-filter/EnumFacetFilter.tsx

```diff
@@ -115,18 +151,71 @@ export const EnumFacetFilter: React.FunctionComponent<EnumFacetFilterProps> = ({
       ></FacetFilterHeader>
       <div style={{ display: isCollapsed ? 'none' : 'block' }}>
         <div className="EnumFacetFilter__checkboxContainer--forAll">
-          <Checkbox
-            className="EnumFacetFilter__checkbox"
-            onChange={() => onClear(columnModel.name)}
-            key="select_all"
-            checked={facetValues.filter(item => item.isSelected).length === 0}
-            label="All"
-            id="select_all"
-          ></Checkbox>
+          <div className={showSearch ? 'EnumFacetFilter__search active' : 'EnumFacetFilter__search'}>
+            <button  // Close Search Button
+              className="EnumFacetFilter__closeSearch"
+              onClick={() => {
+                setFilteredSet(facetValues)
+                setShowSearch(false)
+                setIsShowAll(false)
+              }}
+            >
+              <FontAwesomeIcon
+                className="EnumFacetFilter__previous"
+                icon={faArrowLeft}
+              />
+            </button>
+            <button  // Clear Search Filter Text Button
+              className="EnumFacetFilter__resetSearch"
+              onClick={() => {
+                setSearchText('')
```

Comment:
Should this reset the filtered values to all facet values? if you clear the search then I would think that the filter should reset and show all values

Reply:
No for now, there is a requirement to leave checkbox the same if I clear the search text. But I think I should double check with design.